lua/null-ls/sources.lua

```diff
@@ -42,26 +42,51 @@ local register_source = function(source)
     registered.names[source.name] = true
 end
 
-M.is_available = function(source, filetype, method)
-    if source._disabled or source.generator._failed then
-        return false
+local matches_filetype = function(source, filetype)
+    if not filetype then
+        return true
+    end
+
+    local filetypes = vim.split(filetype, ".", { plain = true })
+    if string.find(filetype, ".") then
```

Comment:
```suggestion
    if string.find(filetype, "%.") then
```

We need to escape this or else it matches any character (so this condition will always be met if `filetype` is not an empty string).

Reply:
Done.